File: core/js/InputParam.js
class InputParam{
  constructor(title,name,defaultValue,userValue,typeName='int',start=0,step=0,stop=0){
    this.title=title; this.name=name; this.defaultValue=defaultValue; this.typeName=typeName;
    this.start=start; this.step=step; this.stop=0; //for optimization
    this.value=userValue?userValue:this.defaultValue;
  }
  static getByName(arr,name){
    for(var i=0;i<arr.length;i++)
      if(arr[i].name==name)return arr[i];
    return null;
  }
}

File: core/py/InputParam.py
class InputParam:
    def __init__(self,title,name,defaultValue,userValue,typeName='int',start=0,step=0,stop=0):
        self.title=title
        self.name=name
        self.defaultValue=defaultValue
        self.userValue=userValue
        self.typeName=typeName
        #optimzations
        self.start=start
        self.step=step
        self.stop=stop
